SlicerLite/SlicerLiteLib/SlicerLiteModuleWidget.py

```diff
@@ -0,0 +1,170 @@
+import qt, slicer
+
+from .ButtonItemDelegate import DeleteButtonItemDelegate, DicomMetadataButtonItemDelegate
+from .DataLoader import DataLoader
+from .EventFilters import DragAndDropEventFilter
+from .UIUtils import UIUtils
+from .SlicerUtils import *
+from .ItemModel import ItemModel, Item
+
+
+class SlicerLiteModuleWidget(qt.QWidget):
+    def __init__(self, parent=None):
+        super(SlicerLiteModuleWidget, self).__init__(parent)
+        qt.QVBoxLayout(self)
+        self._lastOpenedDirectory = ""
+
+        self.data_loader = DataLoader()
+        self.item_table_model = ItemModel()
+        self.item_table_view = qt.QTableView()
+        self.item_table_model.dataChanged.connect(lambda _: self.item_table_view.viewport().update())
+        self.delete_button_item_delegate = DeleteButtonItemDelegate()
+        self.dicom_tags_button_item_delegate = DicomMetadataButtonItemDelegate()
+
+        self.buttons_delegate = [self.delete_button_item_delegate, self.dicom_tags_button_item_delegate]
+
+        # Setup event filter
+        self.filter = DragAndDropEventFilter(slicer.util.mainWindow(), self.load_dicom_directory)
+        slicer.util.mainWindow().installEventFilter(self.filter)
+
+        self.setup_ui()
+
+    def setup_ui(self) -> None:
```

Comment:
Overall review : Type hint for return is very inconsistent in repo and overall return None is not interesting to type hint.